.github/workflows/cicd_rust_main_on_push.yml

```diff
@@ -1,12 +1,13 @@
 name: CICD on main
 
 on:
+  pull_request:
```

Comment:
I think putting it in `main_on_push` file might causes it to publish it when there's PR created.

Reply:
I updated at line 65
This one is fine to run all
But only build and publish the package on push event (after merge successfully)